source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -131,6 +131,12 @@ export abstract class CardElement {
     private _defaultRenderedElementDisplayMode: string = null;
     private _padding: PaddingDefinition = null;
 
+    public elementId: string;
+
+    public constructor(){
+        this.elementId = Utils.guidGenerator();
```

Comment:
I'd rather not do that. I'd rather lead card authors into explicitly specifying Ids where necessary/required. For example, the designer has a warning when an input element doesn't have an id; this prompts the card author to set the id and could lead them to better understand what it's used for.

Reply:
P0: Random ID generation isn't ideal for adaptive - IDs really only necessary for inputs and we would rather educate the developer on what they need to include and when than just get them reliant on autogeneration and not needing it

Reply:
In progress.